backend/src/main/java/com/engineering/thesis/backend/model/authentication/LoginRequest.java

```diff
@@ -3,14 +3,22 @@
 import lombok.Getter;
 import lombok.Setter;
 
+import javax.validation.constraints.Email;
 import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
 
 @Getter
 @Setter
 public class LoginRequest {
-    @NotBlank
+    @NotBlank(message = "Email can't be empty")
+    @Size(max = 50, message = "Email can be up to 50 characters long")
+    @Email(message = "Invalid email address")
     private String email;
 
-    @NotBlank
+    @NotBlank(message = "Password can't be empty")
+    @Size(min = 6, max = 50, message = "Password should be 6 to 50 characters long")
+    @Pattern(regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@#$!%^*?&])[A-Za-z\\d@$!%*?&].*$",
```

Comment:
(?=.*\\d) powoduje że chcesz znaleźć '\' a nie numer 0-9. (?=.*\d) Powinno działać poprawnie.

Reply:
Formatowanie w komentarzu niestety się wysypało w pierwszym regexie \ \  szuka ci \ .

Reply:
`^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@#$!%^*?&])[A-Za-z\\d@$!%*?&].*$` poprawiony regexp

Reply:
Taki format w Stringu powoduje error, i nawet jak przekopiowuję poprawiony regex to Intellij sam poprawia z dodatkowym \, aby był poprawnie sformatowany :) Testowałam i szuka liczby, a nie ukośnika ;)
![illegal_escape_character_in_string_literal](https://user-images.githubusercontent.com/31111330/100221763-6c253e00-2f19-11eb-93f2-505098bd7505.jpg)

Reply:
Jest git można zamykać. @Pattern korzysta ze składni Javy? To wyrażenie dla 8 nie przejdzie jak coś.

Reply:
Jeśli chodzi o oznaczoną osobę to widzę po profilu, że korzysta z JavyScript i Phyton'a xD a jeśli chodzi o tą adnotację, to w końcu jest biblioteka Javy, więc musi korzystać ze składni Javy ;)
A co do tego czy dla jakiejś liczby nie przejdzie to zrobiłam screeny, że przechodzi od 0 do 9 i nie przechodzi tylko jeśli jest brak liczby w haśle:
![0](https://user-images.githubusercontent.com/31111330/100239399-a995c580-2f31-11eb-8f5c-68cb94f63994.jpg)
![1](https://user-images.githubusercontent.com/31111330/100239404-aa2e5c00-2f31-11eb-94f9-65bbfffd8f95.jpg)
![2](https://user-images.githubusercontent.com/31111330/100239408-aac6f280-2f31-11eb-9fa5-64ed1e3f000b.jpg)
![3](https://user-images.githubusercontent.com/31111330/100239414-abf81f80-2f31-11eb-8901-63ea81f8fefd.jpg)
![4](https://user-images.githubusercontent.com/31111330/100239418-ac90b600-2f31-11eb-971e-cbb6fa6bfb21.jpg)
![5](https://user-images.githubusercontent.com/31111330/100239421-ad294c80-2f31-11eb-86f6-ece4f2d65f9a.jpg)
![6](https://user-images.githubusercontent.com/31111330/100239423-ad294c80-2f31-11eb-8f75-5597e215190e.jpg)
![7](https://user-images.githubusercontent.com/31111330/100239425-adc1e300-2f31-11eb-9865-4d5331a4f8af.jpg)
![8](https://user-images.githubusercontent.com/31111330/100239430-ae5a7980-2f31-11eb-9448-13b81ee7dfd1.jpg)
![9](https://user-images.githubusercontent.com/31111330/100239433-aef31000-2f31-11eb-9321-9701752e665a.jpg)
![without_number](https://user-images.githubusercontent.com/31111330/100239435-aef31000-2f31-11eb-90f7-d4cdb8603dbd.jpg)